### PR TITLE
Replace SpatialJoinNode::Expression with RowExpression

### PR DESCRIPTION
**Pull request by highker**
On top of #12578

### CODE REVIEW

**Comment by highker** [comment]
Ready for review

File: presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java [hunk 1]
```diff
@@ -1619,23 +1617,31 @@ public PhysicalOperation visitJoin(JoinNode node, LocalExecutionPlanContext cont
...
                     return operation.get();
                 }
             }
 
-            List<ComparisonExpression> spatialComparisons = extractSupportedSpatialComparisons(filterExpression);
-            for (ComparisonExpression spatialComparison : spatialComparisons) {
-                if (spatialComparison.getOperator() == LESS_THAN || spatialComparison.getOperator() == LESS_THAN_OR_EQUAL) {
+            List<CallExpression> spatialComparisons = extractSupportedSpatialComparisons(filterExpression, metadata.getFunctionManager());
+            for (CallExpression spatialComparison : spatialComparisons) {
+                FunctionMetadata functionMetadata = metadata.getFunctionManager().getFunctionMetadata(spatialComparison.getFunctionHandle());
```
Comment (hellium01) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Use ```FunctionResolution``` instead.

File: presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java [hunk 2]
```diff
@@ -74,6 +97,16 @@ private static boolean isSupportedSpatialFunction(FunctionCall functionCall)
                 .collect(toImmutableList());
     }
 
+    public static List<CallExpression> extractSupportedSpatialComparisons(RowExpression filterExpression, FunctionManager functionManager)
+    {
+        return LogicalRowExpressions.extractConjuncts(filterExpression).stream()
+                .filter(CallExpression.class::isInstance)
+                .map(CallExpression.class::cast)
+                .filter(call -> functionManager.getFunctionMetadata(call.getFunctionHandle()).getOperatorType().map(OperatorType::isComparisonOperator).orElse(false))
```
Comment (hellium01) [inline presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java]: Use ```FunctionResolution``` instead.

File: presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java [hunk 3]
```diff
@@ -152,7 +155,12 @@ private PlanNodeStatsEstimate computeInnerJoinStats(JoinNode node, PlanNodeStats
                 return crossJoinStats;
             }
             // TODO: this might explode stats
-            return filterStatsCalculator.filterStats(crossJoinStats, node.getFilter().get(), session, types);
+            if (isExpression(node.getFilter().get())) {
```
[the remark this answers is not in the document]
Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java]: No it won't; we need to handle both `RowExpression` and `Expression` here.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java (hunk 1 above)
Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: The goal is to get the actual `OperatorType`. `FunctionResolution` won't help.

**Comment by highker** [comment]
Comments addressed; thanks for review.

**Pull request merged**